src/python/grpcio_tests/tests_aio/unit/channel_test.py

```diff
@@ -114,6 +125,22 @@ def test_call_to_the_void(self):
 
         self.loop.run_until_complete(coro())
 
+    def test_unary_unary_metadata(self):
+
+        async def coro():
+            server_target, _ = await start_test_server()  # pylint: disable=unused-variable
+
+            async with aio.insecure_channel(server_target) as channel:
+                hi = channel.unary_unary(
+                    _UNARY_CALL_METHOD,
+                    request_serializer=messages_pb2.SimpleRequest.SerializeToString,
+                    response_deserializer=messages_pb2.SimpleResponse.FromString)
+                call = hi(messages_pb2.SimpleRequest(), metadata=_INVOCATION_METADATA)
+                self.assertIsNotNone(await call.initial_metadata())
+                self.assertIsNotNone(await call.trailing_metadata())
```

Comment:
Correct me if I'm wrong, but we are basically testing the default value here? because the `_INVOCATION_METADATA` is not finally wired in the server-side since `_maybe_echo_metadata` does not work, amb I wrong?

If so, maybe for having a reliable coverage we would need to implement something that allows the code to test that the metadata is received. For that - one solution that does not require the context- is create another gRPC endpoint, for example, `MetadataEcho` which would go through to all of hte metadata and will make an echo of the metadata found as metadata.

Besides that, and in the same way as being done in the `cal_test.py` for the details, I would implement a couple of tests for testing that `initial_metadata` and `trailing_metadata` are _awaitable_, in these tests we could double-check that the value returned is the default value when there is no metada returned by the user.